libs/client-api/src/v2/README.md

```diff
@@ -0,0 +1,929 @@
+    ClientB ->> ClientB: Local edit 1
+    Note over ClientB: SV now: [A:1, B:1]
+    ClientB ->> ClientB: Local edit 2
+    Note over ClientB: SV now: [A:1, B:2]
+    Note over ClientA, Server: 2. Concurrent Edits
+    ClientA ->> Server: ClientMessage::Update
+    Note over Server: SV now: [A:2, B:0]
+    Note over ClientB, Server: 3. Reconnection
+    Note over ClientB: Reconnects
+    ClientB ->> Server: ClientMessage::Manifest {last_message_id, SV: [A:1, B:2]}
+    Note over Server: 4. Reconciliation
+    Note over Server: Compare SVs<br>Missing on client: [A:2]<br>Missing on server: [B:1, B:2]
+    Note over Server, ClientB: 5. Two-way Sync
+    Server -->> ClientB: ServerMessage::Update {update: A:2}
+    ClientB -->> Server: ClientMessage::Update {update: B:1}
+    ClientB -->> Server: ClientMessage::Update {update: B:2}
+    Note over Server, ClientA: 6. Propagation
+    Server ->> ClientA: ServerMessage::Update {update: B:1}
+    Server ->> ClientA: ServerMessage::Update {update: B:2}
+    Note over ClientA, ClientB: All clients synchronized with SV [A:2, B:2]
+```
+
+The offline editing and reconnection process contains these key phases:
+
+1. **Disconnection Phase**:
+    - Client disconnects (intentionally or due to network issues)
+    - Client maintains local document state and continues tracking changes
+    - All edits update the local state vector but are stored in memory
+
+2. **Concurrent Edits Phase**:
+    - While Client B is offline, Client A continues making changes
+    - Server state advances with Client A's updates
+    - State vector divergence occurs between clients
+
+3. **Reconnection Phase**:
+    - On reconnection, client sends a Manifest message with:
+        - Its current state vector [A:1, B:2]
+        - Last received message ID
+    - This triggers the `publish_pending_collabs` mechanism in the client
+
+4. **Reconciliation Phase**:
+    - Server compares client's state vector with its own
+    - Identifies missing updates on both sides
+    - Server discovers it needs to send [A:2] to client
+    - Server discovers it's missing [B:1, B:2] from client
+
+5. **Two-way Synchronization Phase**:
+    - Server sends missing updates to client
+    - Client sends its local changes to server
+    - Using `calculate_missing_update` function on the server
+
+6. **Propagation Phase**:
+    - Server broadcasts newly received updates to all other clients
+    - All clients converge to the same state [A:2, B:2]
+
+The offline editing capability is made possible by:
+
+- Local-first edits with CRDT data structures
+- State vector tracking to identify precisely what updates are missing
+- Two-way synchronization on reconnection
+- The `WorkspaceController` client implementation that manages connection state
+
+## Resilience Mechanisms
+
+The AppFlowy collaboration system implements several mechanisms to ensure reliability and consistency across various
+challenging scenarios.
+
+### Resource ID (Rid) Management
```

Comment:
Actually Rid stands for Redis ID. It's the same as `MessageId` we already have, but my problem was that MessageId is defined in module that is not shared dependency - also MessageId is very generic name ;)